components/bio-formats-plugins/src/loci/plugins/in/ImagePlusReader.java

```diff
@@ -570,59 +579,71 @@ private String constructSliceLabel(int ndx, IFormatReader r,
   {
     r.setSeries(series);
 
-    final int[] zct = r.getZCTCoords(ndx);
-    final StringBuffer sb = new StringBuffer();
-
-    int[] subC;
-    String[] subCTypes;
-    Modulo moduloC = r.getModuloC();
-    if (moduloC.length() > 1) {
-      subC = new int[] {r.getSizeC() / moduloC.length(), moduloC.length()};
-      subCTypes = new String[] {moduloC.parentType, moduloC.type};
-    } else {
-      subC = new int[] {r.getSizeC()};
-      subCTypes = new String[] {FormatTools.CHANNEL};
+    String sliceLabelPattern = LociPrefs.getSliceLabelPattern();
+
+    MetadataStore store = r.getMetadataStore();
+    MetadataRetrieve retrieve = store instanceof MetadataRetrieve ? (MetadataRetrieve) store : new DummyMetadata();
+    String filename = sliceLabelPattern.replaceAll(FormatTools.SERIES_NUM, String.format("%d", series));
+
+    String imageName = retrieve.getImageName(series);
+    if (imageName == null) imageName = "Series" + series;
+    filename = filename.replaceAll(FormatTools.SERIES_NAME, imageName);
+
+    DimensionOrder order = retrieve.getPixelsDimensionOrder(series);
```

Comment:
Unit tests are currently failing, as ```order``` will be null if ```retrieve``` is a ```DummyMetadata```.  Using the reader's dimension order instead should work.